harvester/file.py

```diff
@@ -175,12 +182,20 @@ class UnknownTypeFile(File):
     To be deleted when we figure out the file type detection.
     """
 
+    def __init__(self, checksum, algorithm, location_xlink, binding_dc_identifier):
+        super().__init__(checksum, algorithm, location_xlink, binding_dc_identifier)
+        self.filetype = "UnknownTypeFile"
+
 
 class ALTOFile(File):
     """
     An XML file with contents of a page described using the ALTO schema.
     """
 
+    def __init__(self, checksum, algorithm, location_xlink, binding_dc_identifier):
+        super().__init__(checksum, algorithm, location_xlink, binding_dc_identifier)
+        self.filetype = "ALTOFile"
```

Comment:
Adding variable that is essentially `self.__class__.__name__` feels slightly boilerplate'y. Depending on the expected benefit we get from this in terms of user-friendliness I can come up with two alternative improvements:

Case 1: we really benefit from the simplicity of `file.filetype` syntax, in which case we can add a property to the superclass and determine type there with `__class__.__name__`

Case 2: we expect to do file type checks pretty rarely and/or not in too many places, in which case we can use `isinstance(file, ALTOFile)` instead.

My gut feeling would be that we are closer to case 2, as we are mainly interested in bulk downloads done for all files listed in METS. Further, if we provide wrapper functions such as the current `download_alto_files`, external users never need to worry about e.g. importing `ALTOFile` just to pass it as a parameter.

Reply:
Using `isinstance()` is the best choice by far. Fixing!